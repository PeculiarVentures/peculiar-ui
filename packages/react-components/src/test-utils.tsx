import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
