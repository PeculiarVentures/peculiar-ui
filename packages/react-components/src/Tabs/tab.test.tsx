import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Tab } from './index';

describe('<Tab />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(<Tab id="tab1">Tab-1</Tab>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(<Tab id="tab1" className="my-class-name">Tab-1</Tab>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass disabled', () => {
    const { asFragment } = render(<Tab id="tab1" disabled>Tab-1</Tab>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass "component"', () => {
    const { asFragment } = render(<Tab id="tab1" component="a">Tab-1</Tab>);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('should render colors', () => {
    const colors: Array<React.ComponentProps<typeof Tab>['color']> = [
      'black',
      'white',
    ];
    const selectVariant: boolean[] = [
      true,
      false,
    ];

    colors.forEach((color) => {
      selectVariant.forEach((selected) => {
        it(`colors: "${color}" & selected: "${selected}"`, () => {
          const { asFragment } = render(
            // @ts-ignore
            <Tab
              id="tab1"
              color={color}
              selected={selected}
            >
              Tab-1
            </Tab>,
          );

          expect(asFragment()).toMatchSnapshot();
        });
      });
    });
  });
});
