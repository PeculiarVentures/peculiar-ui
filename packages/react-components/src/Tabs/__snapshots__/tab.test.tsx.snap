// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab /> should pass "component" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled) {
  color: var(--pv-color-gray-10);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

<a
    class="emotion-0"
    id="tab1"
    role="tab"
    type="button"
  >
    <span
      class="emotion-1"
    >
      Tab-1
    </span>
  </a>
</DocumentFragment>
`;

exports[`<Tab /> should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled) {
  color: var(--pv-color-gray-10);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

<button
    class="my-class-name emotion-0"
    id="tab1"
    role="tab"
    type="button"
  >
    <span
      class="emotion-1"
    >
      Tab-1
    </span>
  </button>
</DocumentFragment>
`;

exports[`<Tab /> should pass disabled 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled) {
  color: var(--pv-color-gray-10);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

<button
    class="emotion-0"
    disabled=""
    id="tab1"
    role="tab"
    type="button"
  >
    <span
      class="emotion-1"
    >
      Tab-1
    </span>
  </button>
</DocumentFragment>
`;

exports[`<Tab /> should render colors colors: "black" & selected: "false" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled) {
  color: var(--pv-color-gray-10);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

<button
    class="emotion-0"
    id="tab1"
    role="tab"
    type="button"
  >
    <span
      class="emotion-1"
    >
      Tab-1
    </span>
  </button>
</DocumentFragment>
`;

exports[`<Tab /> should render colors colors: "black" & selected: "true" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: var(--pv-color-secondary);
}

.emotion-0:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled) {
  color: var(--pv-color-black);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

<button
    class="emotion-0"
    id="tab1"
    role="tab"
    type="button"
  >
    <span
      class="emotion-1"
    >
      Tab-1
    </span>
  </button>
</DocumentFragment>
`;

exports[`<Tab /> should render colors colors: "white" & selected: "false" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled) {
  color: var(--pv-color-gray-9);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

<button
    class="emotion-0"
    id="tab1"
    role="tab"
    type="button"
  >
    <span
      class="emotion-1"
    >
      Tab-1
    </span>
  </button>
</DocumentFragment>
`;

exports[`<Tab /> should render colors colors: "white" & selected: "true" 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: var(--pv-color-secondary);
}

.emotion-0:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled) {
  color: var(--pv-color-white);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

<button
    class="emotion-0"
    id="tab1"
    role="tab"
    type="button"
  >
    <span
      class="emotion-1"
    >
      Tab-1
    </span>
  </button>
</DocumentFragment>
`;

exports[`<Tab /> should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: inherit;
  outline: none;
  cursor: pointer;
  box-sizing: border-box;
  border: 0;
  border-bottom: 3px solid transparent;
  min-height: var(--pv-size-base-12);
  height: 100%;
  padding: 0 var(--pv-size-base-3);
  background-color: transparent;
  -webkit-transition: background-color 200ms,color 200ms,border-color 200ms;
  transition: background-color 200ms,color 200ms,border-color 200ms;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:disabled {
  cursor: not-allowed;
  color: var(--pv-color-gray-7);
}

.emotion-0:not(:disabled) {
  color: var(--pv-color-gray-10);
}

.emotion-0:not(:disabled):hover {
  background-color: var(--pv-color-secondary-tint-5);
}

.emotion-0:not(:disabled):focus-visible {
  background-color: var(--pv-color-secondary-tint-4);
}

.emotion-0:not(:disabled):active {
  background-color: var(--pv-color-secondary-tint-3);
}

.emotion-1 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

<button
    class="emotion-0"
    id="tab1"
    role="tab"
    type="button"
  >
    <span
      class="emotion-1"
    >
      Tab-1
    </span>
  </button>
</DocumentFragment>
`;
